tests/test_connection.py

```diff
@@ -4494,13 +4494,19 @@ def test_reconnect_no_additional_transient_failure(node_factory, bitcoind):
 
 @pytest.mark.xfail(strict=True)
 def test_offline(node_factory):
-    # if get_node starts it, it'll expect an address, so do it manually.
-    l1 = node_factory.get_node(options={"offline": None}, start=False)
-    l1.daemon.start()
+    l1, l2 = node_factory.line_graph(2, opts={'may_reconnect': True})
+
+    # restart in offline
+    l1.stop()
+    l1.daemon.opts['offline'] = None
+    l1.start()
 
-    # we expect it to log offline mode an not to create any listener
+    # we expect it to log offline mode
     assert l1.daemon.is_in_log("Started in offline mode!")
+    # not to create any listener
     assert not l1.daemon.is_in_log("connectd: Created listener on")
+    # and not to allow connections
+    assert not l1.rpc.connect(l2.info['id'], 'localhost', l2.port)
```

Comment:
we are not an rpc error here?

Reply:
Yeah likely correct, I'll fix that line on the test, when cln refuses to make connections with offline nodes so I can check for the actual error

Reply:
No, this should work.  Starting a node offline then manually bringing it online should work.  Though in theory plugins should not call connect in this case too (currently, they can!)

Reply:
Absolutely, we use this trick extensively to defer automatic reconnections until we have the signer to perform the handshake. The RPC is rather useful, even if you don't want to connect.

For example, many of my instructions to recover or restore will start exactly with "Start with `--offline`" because it allows us to connect to our peers one by one, and have a controlled way of checking if we suffered dataloss, and not risk incurring loss.

So for me this is a NACK.